BAU: Cleanup implementation of nimbusdsOauth2OidcSdk library

sammyiyke opened the pull request:
<!-- Provide a general summary of your changes in the Title above -->
<!-- Include the Jira ticket number in square brackets as prefix, eg `[P4-XXXX] PR Title` -->

## Proposed changes

Cleans up how the `nimbusdsOauth2OidcSdk` library is implemented

### What changed

<!-- Describe the changes in detail - the "what"-->

Changed how `nimbusdsOauth2OidcSdk` library dependency is exposed consumers of common-services.

- Moves the `nimbusdsOauth2OidcSdk` dependency to the api level configuration in `common-services`

- Removes the `nimbusdsOauth2OidcSdk` dependency from the implementation in lambdas that implement the `common-services` library

### Why did it change

By moving the dependecy to the `api` level configurations the `nimbusdsOauth2OidcSdk` will  be transitively exposed to consumers of the library. This would speed up compilation and makes it cleaner for lambdas as they do not directly implement this libary

Some benefits include:

- Reduces duplication across the dependency declaration
- Faster compilation
- Less recompilations when implementation dependencies change
- Less dependency leaks
- Cleaner publishing

Also helps when new lambdas are created - no need to implement a library that is not directky used 
😅

<!-- Describe the reason these changes were made - the "why" -->

sammyiyke committed:
BAU: Moved nimbusdsOauth2OidcSdk to api level

- Moves the nimbusdsOauth2OidcSdk library to the API level in common services

- Updates lambdas ti use the nimbusdsOauth2OidcSdk from the common-services library when imported

Wynndow reviewed (commented):
I like this, but I think it has the potential to go further. A whole bunch of the other libs (most i think), are using `common-services`, and could possibly also have their direct dependency on nimbus removed

Joe-Edwards-GDS reviewed (commented):
This looks fine, but I think the main benefit is reducing duplication across the dependency declaration, not speed of compilation.

The benefits you list (reduced recompilation etc.) apply when moving dependencies the other way:
- If a library is declared as an 'API' dependency, it becomes a transitive dependency and is required when compiling all other projects that depend upon it
- If a library is declared as an 'Implementation' dependency it is not transitive, and must be explicitly declared in other projects where it's needed

In this case, the old approach requires every lambda to declare whether it needs nimbus to compile, but now we automatically require it (whether it's technically needed or not).

I like reducing duplication though, and basically everything needs all these dependencies.

sammyiyke committed:
BAU: Moved nimbusdsOauth2OidcSdk to api level

- Updates libraries to use the `nimbusdsOauth2OidcSdk` from the `common-services` library when imported
